Implemented new modifiers for mvaTTH Run 3

Cvico opened the pull request:
#### PR description:
## Brief description
Dear reviewers,

This PR is to introduce the newest training of the Muon mvaTTH in CMSSW. The training has been performed following what was done already done for Run 2, but using Run 3 MC samples for training and validation.

## Context of the PR
The goal for this new training is to be used as a substitute of the current mvaTTH **only in Run 3 MC production**. That is: for Run 2 we still need to keep the current mvaTTH implementation (can be found in these lines), and for Run 3 we want to make use of an era modifier to fetch the proper --new -- weights that have been computed.

## What this PR includes
**This change should only affect Run 3 production**. For that, this PR includes:
 1. The required era modifiers in `Configuration/Eras/`.
     * I've added two new era modifiers `run3_muon_2022` and `run3_muon_2023`, which are called from the general `Run3_Eras_cff.py`. 
     * I hope this is the optimal way to handle our use case.
 3. The call to the era modifier and the implementation of the new mvaTTH evaluation.
     * How the mva is evaluated changes a little bit with respect to the previous version: mainly the order in which the variables are parsed to TMVA and also we have added a new additional variable which is `LepGood_pfRelIso03_all`.
     * Which weights should be read in order to properly compute the mva score.

## Validation
I've performed a local test running a simple `cmsRun` script to see that the mva is loaded and nanoAOD is in fact produced. 

## Other comments
Please let me know if the proposal is reasonable, and if there's anything else I can do for testing. I'm adding the L2 muon convener @JanFSchulte so he can follow up the PR as well. Feel free to add any other muon convener (I don't have their github user names unfortunately).

## Linked PRs
This is the link for the PR that includes the weights in CMS-data: [pull16](https://github.com/cms-data/PhysicsTools-NanoAOD/pull/16).

Commit:
Implemented new modifiers for mvaTTH Run 3

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"ba550c090f5a34bf49ecfda0430eb485aaa8d885":{"files":["Configuration/Eras/python/Era_Run3_cff.py","Configuration/Eras/python/Modifier_run3_muon_2022_cff.py","Configuration/Eras/python/Modifier_run3_muon_2023_cff.py","PhysicsTools/NanoAOD/python/muons_cff.py"],"squashed":false,"time":1709674205}},"emoji":{},"last_seen_sha":"ba550c090f5a34bf49ecfda0430eb485aaa8d885","signatures":{}} -->

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44322/39349

- This PR adds an extra 28KB to repository

cmsbuild commented:
A new Pull Request was created by  @Cvico for master.

It involves the following packages:

- Configuration/Eras (**operations**)
- PhysicsTools/NanoAOD (**xpog**)


@davidlange6, @antoniovilela, @rappoccio, @vlimant, @fabiocos, @cmsbuild, @hqucms can you please review it and eventually sign? Thanks.
@makortel, @gpetruc, @AnnikaStein, @missirol, @fabiocos, @Martin-Grunewald this is something you requested to watch as well.
@antoniovilela, @rappoccio, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

vlimant commented:
I'd rather have the default be run3, and you revert to the old using `~run3`

Cvico commented:
> I'd rather have the default be run3, and you revert to the old using `~run3`

Thanks @vlimant for the suggestion. I don't have a feeling on what's the best option here, but could you point me on how to properly do this? 

Do I have to change [these lines](https://github.com/cms-sw/cmssw/blob/ba550c090f5a34bf49ecfda0430eb485aaa8d885/PhysicsTools/NanoAOD/python/muons_cff.py#L89-L110), and call the run2 modifier instead of the run3 modifier?

vlimant commented on PhysicsTools/NanoAOD/python/muons_cff.py:
```diff
@@ -119,6 +119,30 @@
     weightFile = "PhysicsTools/NanoAOD/data/mu_BDTG_2016.weights.xml",
 )
 
+run3_muon.toModify(
```
in line 89, set `muonMVATTH` with the run3 parameters, and there do `run3_common).toModify(` with old paremters. which is just changing the fileinpath, the variable order and remove variables

Cvico replied:
I see, I've updated the config to do this. I was not aware this `~run3` could be done. Let me know if it is what you expected.

Commit:
Setting Run3 as default, included modifier to fetch 2017 weights for Run2

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44322/39359

- This PR adds an extra 28KB to repository

cmsbuild commented:
Pull request #44322 was updated. @antoniovilela, @davidlange6, @vlimant, @fabiocos, @rappoccio, @cmsbuild, @hqucms can you please check and sign again.

JanFSchulte commented:
Hi @Cvico From the POG side we would like to take the opportunity to rename the variable from mvaTTH to mvaPrompt to unify the naming with what is used in the MUO-22-001 paper and to not have it so closely associated with just one analysis since we want this to be an official POG product going forward. Could  you take care of that? It should probably be mentioned in the doc string what the previous name was.

Cvico commented:
> Hi @Cvico From the POG side we would like to take the opportunity to rename the variable from mvaTTH to mvaPrompt to unify the naming with what is used in the MUO-22-001 paper and to not have it so closely associated with just one analysis since we want this to be an official POG product going forward. Could you take care of that? It should probably be mentioned in the doc string what the previous name was.

Hi Jan. From my side I personally like the idea of changing the name to mvaPrompt (more precisely, I don't have strong opinion against it). But since there's a mvaTTH for electrons, I think EGM conveners should be aware of the change as well. I can quickly contact them with the proposal.

Commit:
Updated name from mvaTTH to promptMVA

Cvico commented:
Hi, I've updated the name from  `mvaTTH`  to `promptMVA` as @JanFSchulte suggested, also in the `electrons_cff.py` config. I've checked it produces an output with two branches: `Electron_promptMVA` and `Muon_promptMVA`. 

I've also included two changes: one in `nano_eras_cff.py ` to include the `run3_muon` modifiers, and also I had to modify the muonMVALowPt to fetch the variables for the Run 2 training explicitly. The reason for this latter change is because it was previously working on TOP of the Run 2 mvaTTH configuration, but since we moved this to Run 3 by default, the mvaLowPT had to be explicitly modified with the Run 2 variables.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44322/39392

- This PR adds an extra 36KB to repository

cmsbuild commented:
Pull request #44322 was updated. @davidlange6, @vlimant, @hqucms, @antoniovilela, @fabiocos, @rappoccio, @cmsbuild can you please check and sign again.

RSalvatico commented:
type egamma

AlbertoBelvedere commented on PhysicsTools/NanoAOD/python/electrons_cff.py:
```diff
@@ -359,7 +359,7 @@ def _get_bitmapVIDForEle_docstring(modules,WorkingPoints):
     ),
     externalVariables = cms.PSet(
-        mvaTTH = ExtVar(cms.InputTag("electronMVATTH"),float, doc="TTH MVA lepton ID score",precision=14),
+        promptMVA = ExtVar(cms.InputTag("muonPROMPTMVA"),float, doc="Prompt MVA lepton ID score. Corresponds to the previous mvaTTH",precision=14),
```
Hello! I think here you forgot to change the name from muonPROMPTMVA to electronPROMPTMVA

makortel commented on PhysicsTools/NanoAOD/python/muons_cff.py:
```diff
@@ -86,39 +86,76 @@
 )
 
 run2_muon_2016.toModify(
-    muonMVATTH,
+    muonPROMPTMVA,
     weightFile = "PhysicsTools/NanoAOD/data/mu_BDTG_2016.weights.xml",
 )
```
Given the `~run3_muon` customization below, this `run2_muon_2016` customization is now ineffective in practice.

makortel replied:
In the present setup this `run2_muon_2016` customization is applied to the `muonPROMPTMVA` that IIUC corresponds the Run 3 setup (that has e.g. `LepGood_pfRelIso03_all` variable that was not present in the earlier Run2-based base). Is that the intention?

Commit:
Merged run3_muon_YEAR into one single run3_muon modifier. Addressed remaining comments from PR

Commit:
Put run2_muon_2018 for exclusion in Run3 era config.

Cvico commented:
Hi, comments addressed. I also spotted an issue in the Era Run3 modifier: since run2_muon_2018 was not being explicitly removed there, it was loading the Run 2 weights anyway... Should I also put run2_muon_2017 in that exclusion file? I think now it should be only reading 2022 when the Era config is not run2_muon_2016, 2017 or 2018... But following all this modifiers is a little bit complicated.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44322/39405

- This PR adds an extra 36KB to repository

- Found files with invalid states:
  - Configuration/Eras/python/Modifier_run3_muon_2023_cff.py:
    - Added: ba550c090f5a34bf49ecfda0430eb485aaa8d885
    - Deleted: f8ae9868255f06bf4f627677fe17b1950699d079
  - Configuration/Eras/python/Modifier_run3_muon_2022_cff.py:
    - Added: ba550c090f5a34bf49ecfda0430eb485aaa8d885
    - Deleted: f8ae9868255f06bf4f627677fe17b1950699d079

cmsbuild commented:
Pull request #44322 was updated. @fabiocos, @rappoccio, @vlimant, @hqucms, @cmsbuild, @antoniovilela, @davidlange6 can you please check and sign again.

makortel commented:
> I also spotted an issue in the Era Run3 modifier: since run2_muon_2018 was not being explicitly removed there, it was loading the Run 2 weights anyway...

By quick look excluding `run2_muon_2018` modifier from `Run3` era would be consistent with how `run2_muon201[67]` modifiers are handled in `Run2_201[78]` eras. That change would seem to impact only two places.


----

First case is
https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/NanoAOD/python/triggerObjects_cff.py#L390-L395


AFAICT the `prefiringweight` module is enabled only with
https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/NanoAOD/python/triggerObjects_cff.py#L422-L424
and `_run2_HLTconditions` is specific to the eras `Run2_201[678]`, so the change would be safe.

----

In the second case
https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py#L42
the question is whether these values should apply to `Run3` as well.

----

(on a side note, completely irrelevant to this PR, this customization
https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/NanoAOD/python/triggerObjects_cff.py#L372-L375
violates 6.16 in https://cms-sw.github.io/cms_coding_rules.html#6--packaging-rules-1)

makortel commented:
> Should I also put run2_muon_2017 in that exclusion file?

In what exclusion file?

Cvico commented:
> > Should I also put run2_muon_2017 in that exclusion file?
> 
> In what exclusion file?

I was refering to the `Eras_Run3_cff.py` file where I explicitly excluded the 2018_muon config. But I think inside the 2018_muon config one removes the 2017_muon config, so I don't see need for removing it again

Cvico commented:
> > I also spotted an issue in the Era Run3 modifier: since run2_muon_2018 was not being explicitly removed there, it was loading the Run 2 weights anyway...
> 
> By quick look excluding `run2_muon_2018` modifier from `Run3` era would be consistent with how `run2_muon201[67]` modifiers are handled in `Run2_201[78]` eras. That change would seem to impact only two places.
> 
> First case is
> 
> https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/NanoAOD/python/triggerObjects_cff.py#L390-L395
> 
> AFAICT the `prefiringweight` module is enabled only with
> 
> https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/NanoAOD/python/triggerObjects_cff.py#L422-L424
> 
> and `_run2_HLTconditions` is specific to the eras `Run2_201[678]`, so the change would be safe.
> 
> In the second case
> 
> https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py#L42
> 
> the question is whether these values should apply to `Run3` as well.
> 
> (on a side note, completely irrelevant to this PR, this customization
> 
> https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/NanoAOD/python/triggerObjects_cff.py#L372-L375
> 
> violates 6.16 in https://cms-sw.github.io/cms_coding_rules.html#6--packaging-rules-1)

I think @JanFSchulte @Fedespring might comment better on whether these things are needed for Run 3

makortel commented:
> > > Should I also put run2_muon_2017 in that exclusion file?
> > 
> > 
> > In what exclusion file?
> 
> I was refering to the `Eras_Run3_cff.py` file where I explicitly excluded the 2018_muon config. But I think inside the 2018_muon config one removes the 2017_muon config, so I don't see need for removing it again

Right. The `run2_muon_2017` is already excluded in
https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/Configuration/Eras/python/Era_Run2_2018_cff.py#L24
so excluding it again in something deriving from `Run2_2018` would be confusing.

vlimant commented on Configuration/Eras/python/Modifier_run3_muon_cff.py:
```diff
@@ -0,0 +1,3 @@
+import FWCore.ParameterSet.Config as cms
+
+run3_muon=cms.Modifier()
```
is it still needed in the current implementation ? I don't think so.

Cvico replied:
I think for [this comment](https://github.com/cms-sw/cmssw/pull/44322#issuecomment-1994508415), run3_muon is actually useful. So I'd be inclined to keep it.

vlimant commented on PhysicsTools/NanoAOD/python/muons_cff.py:
```diff
@@ -86,39 +86,76 @@
+muonMVALowPt = muonPROMPTMVA.clone(
     weightFile =  cms.FileInPath("PhysicsTools/NanoAOD/data/mu_BDTG_lowpt.weights.xml"),
     name = cms.string("muonMVALowPt"),
+    variablesOrder = cms.vstring(["LepGood_pt","LepGood_eta","LepGood_jetNDauChargedMVASel","LepGood_miniRelIsoCharged","LepGood_miniRelIsoNeutral","LepGood_jetPtRelv2","LepGood_jetDF","LepGood_jetPtRatio","LepGood_dxy","LepGood_sip3d","LepGood_dz","LepGood_segmentComp"]),
```
how about writing the `variables` and the `variableOrder` only once.
`_muonVariablesOrderLegacy` to be used for `muonMVALowPt` and in the `run2_muon_*` modifiers.
`_muonVariablesLegacy` (with the common variables) `_muonVariablesExt` (with the extra ones ; use `_muonVariablesLegacy+ _muonVariablesExt` in run3 (default config) and the `_muonVariablesLegacy` in the modifiers

Cvico replied:
You mean to add this inside the `muons_cff.py` script directly? Or add this in the run2_muon modifiers?

vlimant replied:
hard to write down what I meant ; in essence with the least re-writing of the same string is preferred. please note #44392 that I created in a view to streamline that module configuration

JanFSchulte commented:
> > > I also spotted an issue in the Era Run3 modifier: since run2_muon_2018 was not being explicitly removed there, it was loading the Run 2 weights anyway...
> > 
> > 
> > By quick look excluding `run2_muon_2018` modifier from `Run3` era would be consistent with how `run2_muon201[67]` modifiers are handled in `Run2_201[78]` eras. That change would seem to impact only two places.
> > First case is
> > https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/NanoAOD/python/triggerObjects_cff.py#L390-L395
> > 
> > AFAICT the `prefiringweight` module is enabled only with
> > https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/NanoAOD/python/triggerObjects_cff.py#L422-L424
> > 
> > and `_run2_HLTconditions` is specific to the eras `Run2_201[678]`, so the change would be safe.
> > In the second case
> > https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py#L42
> > 
> > the question is whether these values should apply to `Run3` as well.
> > (on a side note, completely irrelevant to this PR, this customization
> > https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/NanoAOD/python/triggerObjects_cff.py#L372-L375
> > 
> > violates 6.16 in https://cms-sw.github.io/cms_coding_rules.html#6--packaging-rules-1)
> 
> I think @JanFSchulte @Fedespring might comment better on whether these things are needed for Run 3

For the L1 prefiring corrections, we do not have any new weights for Run 3 (yet), those would be in the work in the L1 group. So excluding the relevant modifications from Run 3 would make sense, since the values are meaningless for Run 3 anyway. I'm not sure what the dummy values filled in that case are, though. 

As for the effective areas for MiniISO, there is no retuning for Run 3, so we would need them to stay as they are for 2018 in all Run 3 Mini/Nano versions.

vlimant commented:
NB #44392

makortel commented:
> As for the effective areas for MiniISO, there is no retuning for Run 3, so we would need them to stay as they are for 2018 in all Run 3 Mini/Nano versions.

Then, given that the present state of this PR excludes the `run2_muon_2018` modifier from the `Run3` era, the
https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py#L40-L42
needs a new line
```py
run3_muon.toModify( process.patMuons, effectiveAreaVec = [0.0566, 0.0562, 0.0363, 0.0119, 0.0064])
```
or you could consolidate
```py
(run2_muon_2017 | run2_muon_2018 | run3_muon).toModify( process.patMuons, effectiveAreaVec = [0.0566, 0.0562, 0.0363, 0.0119, 0.0064])
```

Commit:
Updated effective areas

Cvico commented:
> > As for the effective areas for MiniISO, there is no retuning for Run 3, so we would need them to stay as they are for 2018 in all Run 3 Mini/Nano versions.
> 
> Then, given that the present state of this PR excludes the `run2_muon_2018` modifier from the `Run3` era, the
> 
> https://github.com/cms-sw/cmssw/blob/2e5dc6dfc7a5b5c7fc69b46f79859ee02b97e3b3/PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py#L40-L42
> 
> 
> needs a new line
> ```python
> run3_muon.toModify( process.patMuons, effectiveAreaVec = [0.0566, 0.0562, 0.0363, 0.0119, 0.0064])
> ```
> 
> or you could consolidate
> 
> ```python
> (run2_muon_2017 | run2_muon_2018 | run3_muon).toModify( process.patMuons, effectiveAreaVec = [0.0566, 0.0562, 0.0363, 0.0119, 0.0064])
> ```

Fixed. I opted for the first option.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44322/39468

- This PR adds an extra 28KB to repository

- Found files with invalid states:
  - Configuration/Eras/python/Modifier_run3_muon_2023_cff.py:
    - Added: ba550c090f5a34bf49ecfda0430eb485aaa8d885
    - Deleted: f8ae9868255f06bf4f627677fe17b1950699d079
  - Configuration/Eras/python/Modifier_run3_muon_2022_cff.py:
    - Added: ba550c090f5a34bf49ecfda0430eb485aaa8d885
    - Deleted: f8ae9868255f06bf4f627677fe17b1950699d079

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/Eras/python/Era_Run3_cff.py  modified in PR(s): #44368
  - File  PhysicsTools/NanoAOD/python/electrons_cff.py  modified in PR(s): #44392
  - File  PhysicsTools/NanoAOD/python/muons_cff.py  modified in PR(s): #44392
  - File  PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py  modified in PR(s): #44316, #44335, #44383

jfernan2 commented:
please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests RelVals RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/summary.html
**COMMIT**: bafee581a21ce3c2b0cab672d1b1f6bb838c08c5
**CMSSW**: CMSSW_14_1_X_2024-03-13-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/44322/38125/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 2 errors in the following unit tests:

<pre>---> test TestConfigDP_7 had ERRORS
---> test TestConfigDP_8 had ERRORS
</pre>

## RelVals

- [136.7611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/136.7611_RunJetHT2016EreMINIAOD)```136.7611_RunJetHT2016EreMINIAOD/step2_RunJetHT2016EreMINIAOD.log```
- [136.8311](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/136.8311_RunJetHT2017FreMINIAOD)```136.8311_RunJetHT2017FreMINIAOD/step2_RunJetHT2017FreMINIAOD.log```
- [136.88811](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/136.88811_RunJetHT2018DreMINIAODUL)```136.88811_RunJetHT2018DreMINIAODUL/step2_RunJetHT2018DreMINIAODUL.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [4.53](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/4.53_RunPhoton2012B)
- [139.001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/139.001_RunMinimumBias2021)
- [140.023](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/140.023_RunZeroBias2022B)
- [136.731](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/136.731_RunSinglePh2016B)
- [140.043](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/140.043_RunZeroBias2022C)
- [136.793](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/136.793_RunDoubleEG2017C)
- [140.063](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/140.063_RunZeroBias2022D)
- [136.874](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/136.874_RunEGamma2018C)
- [158.01](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/158.01_HydjetQB12ppRECOin2018reMINIAOD)
- [135.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/135.4_ZEEFS_13)
- [1306.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/1306.0_SingleMuPt1_UP15)
- [1330.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/1330.0_ZMM_13)
- [9.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/9.0_Higgs200ChargedTaus)
- [25.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/25.0_TTbar)
- [141.042](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/141.042_RunZeroBias2023D)
- [312.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/312.0_Pyquen_ZeemumuJets_pt10_2760GeV_2022)
- [13234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/13234.0_TTbar_14TeV+2021FS)
- [14034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/14034.0_TTbar_14TeV+2023FS)
- [14234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/14234.0_TTbar_14TeV+2023FSPU)
- [141.044](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/141.044_RunJetMET2023D)
- [141.046](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/141.046_RunEGamma2023D)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/11634.0_TTbar_14TeV+2021)
- [25202.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/25202.0_TTbar_13)
- [2500.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/2500.4_NANOmc132X)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/12434.0_TTbar_14TeV+2023)
- [12834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/12834.0_TTbar_14TeV+2024)
- [12846.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/12846.0_ZEE_14+2024)
- [12834.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/12834.7_TTbar_14TeV+2024_trackingMkFit)
- [10224.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/10224.0_TTbar_13+2017PU)
- [24900.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/24900.0_CloseByPGun_CE_H_Coarse_Scint+2026D98)
- [24896.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/24896.0_CloseByPGun_CE_E_Front_120um+2026D98)
- [13034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/13034.0_TTbar_14TeV+2024PU)
- [24834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/24834.0_TTbar_14TeV+2026D98)
- [23234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/23234.0_TTbar_14TeV+2026D94)
- [250202.181](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/250202.181_TTbar13TeVPUppmx2018)
- [25034.999](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/25034.999_TTbar_14TeV+2026D98PU_PMXS1S2PR)
- [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrix-results/24834.911_TTbar_14TeV+2026D98_DD4hep)
</details>



## RelVals-INPUT

- [4.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/4.6_MinimumBias2010A)```4.6_MinimumBias2010A/step2_MinimumBias2010A.log```
- [136.72411](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/136.72411_RunJetHT2016B_reminiaodUL)```136.72411_RunJetHT2016B_reminiaodUL/step2_RunJetHT2016B_reminiaodUL.log```
- [136.72412](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/136.72412_RunJetHT2016B_reminiaodUL)```136.72412_RunJetHT2016B_reminiaodUL/step2_RunJetHT2016B_reminiaodUL.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [136.76111](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/136.76111_RunJetHT2016EreMINIAODUL)
- [136.7611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/136.7611_RunJetHT2016EreMINIAOD)
- [136.7721](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/136.7721_RunJetHT2016HreMINIAOD)
- [136.77211](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/136.77211_RunJetHT2016HreMINIAODUL)
- [136.83111](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/136.83111_RunJetHT2017FreMINIAODUL)
- [136.8311](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/136.8311_RunJetHT2017FreMINIAOD)
- [136.88811](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/136.88811_RunJetHT2018DreMINIAODUL)
- [136.9](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/136.9_RunDoubleMuon2016C)
- [140.201](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/140.201_RunJetMET2022D_reMINI)
- [140.202](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/140.202_RunJetMET2022D_reMINI)
- [140.5611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/140.5611_RunHI2018reMINIAOD)
- [158.01](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/158.01_HydjetQB12ppRECOin2018reMINIAOD)
- [159.01](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/159.01_HydjetQ_reminiaodPbPb2022_INPUT)
- [1325.5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/1325.5_ZEE13reMINIAOD)
- [1325.51](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/1325.51_TTbar13reMINIAOD)
- [1325.516](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/1325.516_TTbar13reMINIAODin2016ULpreVFP)
- [1325.5161](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/1325.5161_TTbar13reMINIAODin2016ULpostVFP)
- [1325.518](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/1325.518_TTbar13reMINIAODin2018UL)
- [1325.517](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/1325.517_TTbar13reMINIAODin2017UL)
- [134.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/134.0_MinimumBiasBS)
- [144.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/144.6_MinimumBias2010A)
- [1002.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/1002.0_RRD)
- [13234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/13234.0_TTbar_14TeV+2021FS)
- [13434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/13434.0_TTbar_14TeV+2021FSPU)
- [14034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/14034.0_TTbar_14TeV+2023FS)
- [14234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/14234.0_TTbar_14TeV+2023FSPU)
- [2500.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.3_NANOmc130X)
- [2500.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.1_NANOmc122Xrun3)
- [2500.301](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.301_EGMNANOmc130X)
- [2500.2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.2_NANOmc124Xrun3)
- [2500.211](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.211_NANOdata124Xrun3)
- [2500.21](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.21_NANOdata124Xrun3)
- [2500.311](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.311_NANOdata130Xrun3)
- [2500.31](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.31_NANOdata130Xrun3)
- [2500.314](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.314_muPOGNANO_data13.0)
- [2500.401](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.401_muPOGNANO_mc13.2)
- [2500.315](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.315_EGMNANOdata130Xrun3)
- [2500.316](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.316_lepTimeLifeNANO_data13.0)
- [2500.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.4_NANOmc132X)
- [2500.402](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.402_EGMNANOmc132X)
- [2500.403](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.403_lepTimeLifeNANO_mc13.2)
- [2500.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.0_NANOmc106Xul16v2)
- [2500.01](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae026/38125/runTheMatrixINPUT-results/2500.01_NANOdata106Xul16v2)
</details>

Commit:
Fix import

Cvico commented:
Hi, I fixed the import that failed. Apologies for that rookie mistake.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44322/39483

- This PR adds an extra 48KB to repository

- Found files with invalid states:
  - Configuration/Eras/python/Modifier_run3_muon_2023_cff.py:
    - Added: ba550c090f5a34bf49ecfda0430eb485aaa8d885
    - Deleted: f8ae9868255f06bf4f627677fe17b1950699d079
  - Configuration/Eras/python/Modifier_run3_muon_2022_cff.py:
    - Added: ba550c090f5a34bf49ecfda0430eb485aaa8d885
    - Deleted: f8ae9868255f06bf4f627677fe17b1950699d079

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/Eras/python/Era_Run3_cff.py  modified in PR(s): #44368
  - File  PhysicsTools/NanoAOD/python/electrons_cff.py  modified in PR(s): #44392
  - File  PhysicsTools/NanoAOD/python/muons_cff.py  modified in PR(s): #44392
  - File  PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py  modified in PR(s): #44316, #44335, #44383

cmsbuild commented:
Pull request #44322 was updated. @mandrenguyen, @fabiocos, @antoniovilela, @cmsbuild, @rappoccio, @hqucms, @jfernan2, @davidlange6, @vlimant can you please check and sign again.

Cvico commented:
Hi, I see #44392 has been merged. How do I properly resolve the conflicts? Do I pull the commit and then redo my changes from there?

vlimant commented:
hello, the only conflict is in muon_cff, and the change to BaseMVAValueMapProducer might actually help with this PR. 
the "fun" will begin for the backport, if backport at all, since it is going to be different. I don't think I want just yet to backport #44392

JanFSchulte commented:
If at all possible, we would really like to have this back-ported to 14_0_X so we can have the new training in the Nano for 2024 data and the 2022/2023 ReReco.

JanFSchulte commented:
Ah, and @Cvico, the cleanest solution would be to rebase your branch to the latest IB, following https://cms-sw.github.io/tutorial-resolve-conflicts.html

Cvico commented:
Apologies for the low experience on rebasing:
What should be the new base? I'm doing `git cms-rebase-topic cvico:mvatth_run3_14_1_0_pre0` from `14_1_0_pre0` but it does not find any conflict...

JanFSchulte commented:
You need to rebase with a CMSSW build that has the PR that created the conflict already compiled in. There are twice-daily integration builds that reflect the current state of the repository. In this case, you need to start from a fresh checkout of CMSSW_14_1_X_2024-03-19-1100 and then rebase your branch on that.

vlimant commented:
Hi @Cvico , if you have negative experience with rebasing, as this can get complicated (but a good experience for sft dev), I can take care of cherrypicking your changes and do the necessary in muons_cff in a new PR ; let me know how it goes on your end.
In the meatime, I'll backport #44392 too, to make it easier to get your feature in 14.0.

vlimant commented:
I made #44503 to simplify this.

vlimant commented:
pleas close

Cvico commented:
> Hi @Cvico , if you have negative experience with rebasing, as this can get complicated (but a good experience for sft dev), I can take care of cherrypicking your changes and do the necessary in muons_cff in a new PR ; let me know how it goes on your end. In the meatime, I'll backport #44392 too, to make it easier to get your feature in 14.0.

Hi @vlimant, sorry for the late reply, yesterday I was busy with teaching. I'm not sure what's the best option here, I have no experience with it, I'm not sure if (for time sake) it would be better to fix in another PR.

vlimant commented:
please close
